packages/kbn-expandable-flyout/src/components/context.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { noop } from 'lodash/fp';
+import type { FlyoutLayout } from '../models/layout';
+import type { FlyoutPanel } from '../models/panel';
+import { FlyoutState, initialFlyoutState } from '../models/state';
+import {
+  closeFlyout,
+  closeFlyoutLeftPanel,
+  closeFlyoutPreviewPanel,
+  closeFlyoutRightPanel,
+  openFlyout,
+  openFlyoutLeftPanel,
+  openFlyoutPreviewPanel,
+  openFlyoutRightPanel,
+  previousFlyoutPreviewPanel,
+  selectFlyoutLayout,
+} from '../utils/helpers';
+
+export interface ExpandableFlyoutContext {
+  /**
+   * Right, left and preview panels
+   */
+  panels: FlyoutLayout;
+  /**
+   * Open the flyout with left, right and/or preview panels
+   */
+  openFlyout: (panels: { left?: FlyoutPanel; right?: FlyoutPanel; preview?: FlyoutPanel }) => void;
```

Comment:
nit: would it make sense to have params here as a named type?

Reply:
I thought about it but felt like because it's used only here I didn't want to extract it...